pennylane/beta/queuing/operation.py

```diff
@@ -13,47 +13,154 @@
-    """Container class representing tensor products of observables.
 
-    To create a tensor, simply initiate it like so:
+def operation_inv(self):
+    """Monkeypatched :meth:`~.Operation.inv` method.
 
-    >>> T = Tensor(qml.PauliX(0), qml.Hermitian(A, [1, 2]))
+    Rather than updating some internal attribute, this monkeypatched
+    method instead annotates the queue with the current 'inverse'
+    status.
 
-    You can also create a tensor from other Tensors:
+    This operation acts as a 'radio button', swapping the current
+    boolean property of the 'inverse' annotation on the object in the queue.
+    """
+    current_inv = QueuingContext.get_info(self).get("inverse", False)
+    QueuingContext.update_info(self, inverse=not current_inv)
+    return self
 
-    >>> T = Tensor(T, qml.PauliZ(4))
 
-    The ``@`` symbol can be used as a tensor product operation:
+def operation_expand(self):
+    """Monkeypatched :meth:`~.Operation.expand` method for operations.
 
-    >>> T = qml.PauliX(0) @ qml.Hadamard(2)
-    """
+    Currently, this monkeypatched expand method simply mirrors the
+    existing :meth:`~.Operation.decomposition` method; however with
+    two main differences:
 
-    # pylint: disable=abstract-method,too-few-public-methods
+    * It returns a tape containing the decomposed operations, rather
+      than a list.
 
-    def __init__(self, *args):
-        super().__init__(*args)
-        self.queue()
+    * If a decomposition is not available, it simply returns itself.
 
-    def queue(self):
-        """Queues the Tensor instance and updates the ownership related info if applicable."""
-        qml.QueuingContext.append(self, owns=tuple(self.obs))
+    Returns:
+        .QuantumTape: Returns a quantum tape that contains the
+        operations decomposition, or if not implemented, simply
+        the operation itself.
+    """
+    tape = qml.beta.tapes.QuantumTape()
 
+    with tape:
         try:
-            for o in self.obs:
-                qml.QueuingContext.update_info(o, owner=self)
+            self.decomposition(*self.data, wires=self.wires)
         except NotImplementedError:
-            pass
+            self.__class__(*self.data, wires=self.wires)
+
+    if not self.data:
+        # original operation has no trainable parameters
+        tape.trainable_params = {}
+
+    return tape
+
+
+def tensor_init(self, *args):
+    """Monkeypatched :meth:`~.Tensor.__init__` method, allowing
+    Tensors to queue themselves to the beta :class:`~.QueuingContext`,
+    and to annotate the queue specifying previously queued objects they
+    'own'.
+    """
+    self._eigvals_cache = None
+    self.obs = []
+
+    for o in args:
+        if isinstance(o, qml.operation.Tensor):
+            self.obs.extend(o.obs)
+
+        elif isinstance(o, qml.operation.Observable):
+            self.obs.append(o)
+
+        QueuingContext.update_info(o, owner=self)
+
+    QueuingContext.append(self, owns=tuple(args))
```

Comment:
Once the refactor is complete, this logic will be moved to `Tensor.__init__`, and the monkeypatching logic will be completely removed. Similarly for the other patches in this file.